sycl/source/detail/device_impl.cpp

```diff
@@ -19,15 +19,30 @@ device_impl::device_impl()
     : MIsHostDevice(true),
       MPlatform(std::make_shared<platform_impl>(platform_impl())) {}
 
+device_impl::device_impl(device_interop_handle_t InteropDeviceHandle,
+                         const plugin &Plugin)
+    : device_impl(InteropDeviceHandle, nullptr, nullptr, Plugin) {}
+
 device_impl::device_impl(RT::PiDevice Device, PlatformImplPtr Platform)
-    : device_impl(Device, Platform, Platform->getPlugin()) {}
+    : device_impl(nullptr, Device, Platform, Platform->getPlugin()) {}
 
 device_impl::device_impl(RT::PiDevice Device, const plugin &Plugin)
-    : device_impl(Device, nullptr, Plugin) {}
+    : device_impl(nullptr, Device, nullptr, Plugin) {}
 
-device_impl::device_impl(RT::PiDevice Device, PlatformImplPtr Platform,
+device_impl::device_impl(device_interop_handle_t InteropDeviceHandle,
+                         RT::PiDevice Device, PlatformImplPtr Platform,
```

Comment:
Didn't get the idea. Why do we need two "native" handles here? Why having one is not enough?

Reply:
Please, correct me if I'm wrong, but RT::PiDevice can be OpenCL handle(cl_device_id) only which is exactly native handle you are adding support for.

Reply:
This is essentially an effort to handle BE specific native handles in the future. For OpenCL handles this is a redundant change.
Let's say if we replace the opencl_handle with a cuda native handle of type cuDevice_ . Then cuda plugin can change the API cuda_piextDeviceInterop() to return PiDevice_Argument.get() which returns the cuDevice_ handle from the PiDevice structure in pi_cuda.hpp.
Currently, the sycl APIs only support interoperability with OpenCL, but when this is changed to any BE specific handle, this API will help. I'm not sure if I addressed your concern. Let me know if you were asking something else.

Reply:
Overall agree with @garimagu , different backends need to be able to retrieve the underlying native handle object in a backend-specific way

Reply:
We discussed offline. I was just confused why we have a c'tor which takes two pi and native handle.